hazelcast/include/hazelcast/client/protocol/codec/builtin/list_cn_fixed_size_codec.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <vector>
+#include <boost/optional.hpp>
+
+#include "hazelcast/util/export.h"
+#include "hazelcast/client/protocol/ClientMessage.h"
+
+namespace hazelcast {
+namespace client {
+namespace protocol {
+namespace codec {
+namespace builtin {
+class HAZELCAST_API list_cn_fixed_size_codec
+{
+public:
+    template<typename T>
+    static std::vector<boost::optional<T>> decode(ClientMessage& msg)
+    {
+        msg.skip_frame_header_bytes();
+
+        auto type = msg.get<byte>();
+        auto count0 = msg.get<int32_t>();
+        assert(count0 >= 0);
+        auto count = static_cast<std::size_t>(count0);
```

Comment:
Why are you casting `count0` to `size_t`?

Reply:
because, otherwise compiler generates warning.